design/authenticators/authenticator_management_improvement.md

```diff
@@ -0,0 +1,247 @@
+
+The proposed solution is to add a few simple Conjur CLI commands (and possibly APIs, see two alternatives below), that would hide the complexity and provide a single step to perform each action:
+
+### Add Authenticator
+
+#### Functionality
+
+Defines a new authenticator in Conjur. Behind the scenes, this command will creat the authenticator policy and fill it with all its required data. If the data is not required to be given by the user, like the `authn-k8s` CA cert and key, the information will be generated automatically.
+
+#### Synopsis
+
+conjur config authenticator add *authenticator-type/service-id* [*parameters*]
+
+#### Example
+
+```shell script
+$ conjur config authenticator add authn-oidc/my-idp --provider-uri my-uri --id-token-user-property prop
+```
+
+### Delete Authenticator
+
+#### Functionality
+
+Deletes an existing authenticator in Conjur. Deletes the authenticator policy.
+
+#### Synopsis
+
+conjur config authenticator delete *authenticator-type/service-id*
+
+#### Example
+
+```shell script
+$ conjur config authenticator delete authn-oidc/my-idp
+```
+
+### Update Authenticator
+
+#### Functionality
+
+Updates an existing authenticator in Conjur. Updates the authenticator variables.
+
+#### Synopsis
+
+conjur config authenticator update *authenticator-type/service-id* [*parameters*]
+
+#### Example
+
+```shell script
+$ conjur config authenticator update authn-oidc/my-idp --provider-uri new-uri
+```
+
+### Enable Authenticator
+
+#### Functionality
+
+Enables an authenticator in Conjur.
+
+#### Synopsis
+
+conjur config authenticator enable *authenticator-type/service-id*
+
+#### Example
+
+```shell script
+$ conjur config authenticator enable authn-oidc/my-idp
+```
+
+### Disable Authenticator
+
+#### Functionality
+
+Disables an authenticator in Conjur.
+
+#### Synopsis
+
+conjur config authenticator disable *authenticator-type/service-id*
+
+#### Example
+
+```shell script
+$ conjur config authenticator disable authn-oidc/my-idp
+```
+
+## Backwards Compatibility
+
+- The authenticators keep their same structure in Conjur. Users could still manage them in the way it's done today.  
+- These CLI commands are uniting policy loads and variable updates into a single step, therefore running the commands requires permissions to both load a policy under the `conjur` policy and to updated the variables in that policy.
+
+## Performance
+
+The proposal only simplifies the steps, not adding or changing them, therefore the performance remain the same.
+
+## Security
+
+No security implications. These new CLI commands perform the same action, with an easier experience.
+
+## Documentation
+
+We will need to update the docs of every authenticator, to specify how to use the new CLI commands. In addition, we will need to add documentation for enabling/disabling authenticators.
+
+## Drawbacks
+
+Every new authenticator that would be created for Conjur, should now have a corresponding Conjur CLI command support as
```

Comment:
why? Isn't this CLI command generic? All it needs to do is create a policy and add some variables to it.

Reply:
Because if it's generic, it doesn't guide the users to input the right information and it prevents them from misconfiguring. I think there's big value in that user experience.

Reply:
what will the non-generic implemntation include? Only required parameters?